packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -0,0 +1,183 @@
+    }
+
+    // Wait until the application is stable.
+    await whenStable(applicationRef);
+
+    const injector = applicationRef.injector;
+    const router = injector.get(Router);
+
+    const baseHref =
+      injector.get(APP_BASE_HREF, null, { optional: true }) ??
+      injector.get(PlatformLocation).getBaseHrefFromDOM();
+
+    if (router.config.length === 0) {
+      // No routes found in the configuration.
+      yield { route: baseHref };
+    } else {
+      const compiler = injector.get(Compiler);
+
+      // Retrieve all routes from the Angular router configuration.
+      yield* traverseRoutesConfig(router.config, compiler, injector, baseHref, baseHref);
+    }
+  } finally {
+    platformRef.destroy();
+  }
+}
+
+/**
+ * Recursively traverses the Angular router configuration to retrieve routes.
+ *
+ * Iterates through the router configuration, yielding each route along with its potential
+ * redirection or error status. Handles nested routes and lazy-loaded child routes.
+ *
+ * @param routes - The array of route configurations to process.
+ * @param compiler - The Angular compiler used to compile route modules.
+ * @param parentInjector - The parent injector for lazy-loaded modules.
+ * @param parentRoute - The parent route path to prefix child routes.
+ * @param baseHref - A string that represents the base URL of the application. This is used to correctly
+ *                   resolve relative paths within the route configuration.
+ * @returns An async iterator yielding `RouteResult` objects.
+ */
+async function* traverseRoutesConfig(
+  routes: Route[],
+  compiler: Compiler,
+  parentInjector: Injector,
+  parentRoute: string,
+  baseHref: string,
+): AsyncIterableIterator<RouteResult> {
+  for (const route of routes) {
+    const { path, redirectTo, loadChildren, children } = route;
+
+    if (path === undefined) {
+      // Skip routes without a path.
+      continue;
+    }
+
+    const currentRoutePath = joinUrlParts(parentRoute, path);
+    let normalizedRedirectTo = undefined;
+
+    if (typeof redirectTo === 'string') {
+      if (redirectTo[0] === '/') {
+        normalizedRedirectTo = joinUrlParts(baseHref, redirectTo);
+      } else {
+        // Resolve relative redirectTo based on the current route path
+        const segments = currentRoutePath.split('/');
+        segments.pop(); // Remove the last segment to make it relative
+        normalizedRedirectTo = joinUrlParts(...segments, redirectTo);
+      }
+    }
+
+    yield {
+      route: currentRoutePath,
+      redirectTo: normalizedRedirectTo,
+    };
+
+    if (children?.length) {
+      // Recursively process child routes.
+      yield* traverseRoutesConfig(children, compiler, parentInjector, currentRoutePath, baseHref);
+    }
+
+    if (loadChildren) {
+      // Load and process lazy-loaded child routes.
+      const loadedChildRoutes = await loadChildrenHelper(
+        route,
+        compiler,
+        parentInjector,
+      ).toPromise();
```

Comment:
Since Angular still supports version RxJs 6, we need to use `toPromise`, `firstValueFrom` is only available from RxJs version 7.

It might be worth discussing with the framework team to determine when we plan to drop support for RxJs 6, especially since it is not used in G3

Reply:
I had a feeling RxJS 6 was the reason. We can ask about the current status but I suspect nothing is meaningfully changed for now.